tests/users/test_data.py

```diff
@@ -120,115 +212,117 @@ async def test_administrator(
         )
 
     async def test_force_reset(
-        self,
-        data_layer: DataLayer,
-        fake2: DataFaker,
-        mongo: Mongo,
+        self, data_layer: DataLayer, fake2: DataFaker, mongo: Mongo, pg: AsyncEngine
     ):
         """
         Test that setting and unsetting ``force_reset`` works as expected.
         """
         user = await fake2.users.create()
 
-        assert await get_one_field(mongo.users, "force_reset", user.id) is False
+        assert user.force_reset is False
+        async with (AsyncSession(pg) as session):
+            row = await session.get(SQLUser, 1)
+
+        assert row.force_reset is False
 
         user = await data_layer.users.update(
             user.id, UpdateUserRequest(force_reset=True)
         )
 
+        async with (AsyncSession(pg) as session):
+            row = await session.get(SQLUser, 1)
+
         assert user.force_reset is True
-        assert await get_one_field(mongo.users, "force_reset", user.id) is True
+        assert row.force_reset is True
 
         user = await data_layer.users.update(
             user.id, UpdateUserRequest(force_reset=False)
         )
 
+        async with (AsyncSession(pg) as session):
+            row = await session.get(SQLUser, 1)
+
         assert user.force_reset is False
-        assert await get_one_field(mongo.users, "force_reset", user.id) is False
+        assert row.force_reset is False
 
+    @pytest.mark.parametrize("groups", [0, 1, 2])
     async def test_set_groups(
         self,
         data_layer: DataLayer,
         fake2: DataFaker,
-        mongo: Mongo,
+        pg: AsyncEngine,
         snapshot: SnapshotAssertion,
+        groups,
     ):
         """
         Test that setting ``groups`` works as expected.
         """
-        group_1 = await fake2.groups.create()
-        group_2 = await fake2.groups.create()
-
         user = await fake2.users.create()
```

Comment:
I would be tempted always assign a group to the user to start. That way you can test removal of groups as well as addition